[docs][charts] Add donut chart as a special case of a pie chart (#18652)
mui/mui-x: docs/data/charts/pie/pie.md

{{"demo": "BasicPie.js"}}

> ## Donut chart

Author: Used "donut chart" as title as it is more common than "doughnut chart" according to Google trends. 

Still, normal text is still indexed by our search engine, so "doughnut chart" will also show a match.

> A donut chart (or doughnut chart) is essentially a pie chart with a hole in the middle.

Reviewer: ```suggestion
A donut chart (or doughnut chart) is essentially a pie chart with a hollow center.
```

Author: Much better, thanks 👍
